### PR TITLE
DEX 1065 - Update dependencies

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -24,7 +24,8 @@ Seq(
   // dependencyCheck to check against OWASP
   "net.vonbuchholtz" % "sbt-dependency-check" % "1.3.3",
   // dependencyUpdates
-  "com.timushev.sbt" % "sbt-updates" % "0.5.0"
+  "com.timushev.sbt" % "sbt-updates" % "0.5.0",
+  "com.github.tkawachi" % "sbt-repeat" % "0.1.0"
```
Comment (ngadiyak) [inline project/plugins.sbt:28]: ok or remove this line?

Comment (vsuharnikov) [reply project/plugins.sbt:28]: Right, this plugin should be removed

File: build.sbt [hunk 2]
```diff
@@ -101,9 +101,9 @@ lazy val root = (project in file("."))
 
 inScope(Global)(
   Seq(
-    scalaVersion := "2.13.3",
+    scalaVersion := "2.13.4",
     semanticdbEnabled := true,
-    semanticdbVersion := scalafixSemanticdb.revision,
+    semanticdbVersion := "4.4.7",
```
Comment (vsuharnikov) [inline build.sbt:106]: `scalafixSemanticdb` this version should be updated automatically with a new scalafix version. Do we really need to change this?

File: dex-it/src/test/scala/com/wavesplatform/it/WsSuiteBase.scala [hunk 3]
```diff
@@ -56,6 +56,7 @@ trait WsSuiteBase extends MatcherSuiteBase with HasWebSockets with DiffMatcherWi
           .partitionMap {
             case x: WsError => x.asLeft
             case x: WsPingOrPong => x.asRight
+            case _ => throw new IllegalArgumentException(s"Unexpected error")
```
Comment (vsuharnikov) [inline dex-it/src/test/scala/com/wavesplatform/it/WsSuiteBase.scala:59]: `s` is not required here, only when we substitute an expression. So `"s` could be replaced with `"` :)

File: dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsPingPongBaseSuite.scala [hunk 4]
```diff
@@ -136,7 +136,10 @@ abstract class WsPingPongBaseSuite extends WsSuiteBase {
       }
 
       val expectedError = InvalidJson(Nil)
-      val List(actualInvalidJsonError, actualPongTimeoutError) = errors
+      val (actualInvalidJsonError, actualPongTimeoutError) = errors match {
+        case List(actualInvalidJsonError, actualPongTimeoutError) => (actualInvalidJsonError, actualPongTimeoutError)
+        case _ => throw new IllegalArgumentException(s"Unexpected error")
```
Comment (vsuharnikov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsPingPongBaseSuite.scala:141]: Same. Also there is a simple rule for exceptions I always use:
1. If we are inside a function and we got an invalid argument, we throw `IllegalArgumentException` (if throwing is acceptable)
2. If we outside the function is should be another exception type, e.g. `RuntimeException` which is totally fine here

File: dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala [hunk 5]
```diff
@@ -87,7 +87,10 @@ object TankGenerator {
       .shuffle(
         assets
           .combinations(2)
-          .map { case List(aa, pa) => if (aa >= pa) (aa, pa) else (pa, aa) }
+          .map {
+            case List(aa, pa) => if (aa >= pa) (aa, pa) else (pa, aa)
+            case _ => throw new IllegalArgumentException(s"Can't create asset-pair")
```
Comment (vsuharnikov) [inline dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala:92]: Same + RuntimeException

File: dex-load/src/main/scala/com/wavesplatform/dex/load/request/Request.scala [hunk 6]
```diff
@@ -49,6 +49,7 @@ case class Request(
     pw.println(httpType match {
       case RequestType.POST => mkPost(jsonBody, path, tag, stringBody)
       case RequestType.GET => mkGet(path, tag, headers)
+      case _ => throw new IllegalArgumentException(s"Unexpected request type")
```
Comment (vsuharnikov) [inline dex-load/src/main/scala/com/wavesplatform/dex/load/request/Request.scala:52]: Let's write the httpType for easier debugging:
```scala
case _ => throw new IllegalArgumentException(s"Unexpected request type: $httpType")
```

File: dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala [hunk 7]
```diff
@@ -37,6 +37,7 @@ class WsCollectChangesClient(apiUri: String, address: String, aus: String, obs:
       }
       orderBookUpdates.put(diff.assetPair, updatedOb)
       none
+    case _ => throw new IllegalArgumentException(s"Unexpected message")
```
Comment (vsuharnikov) [inline dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala:40]: Let's write the message for easier debugging:
```scala
case x => throw new IllegalArgumentException(s"Unexpected message: $x")
```

File: dex/src/test/scala/com/wavesplatform/dex/actors/MatcherActorSpecification.scala [hunk 8]
```diff
@@ -501,6 +504,7 @@ object MatcherActorSpecification {
     p.setAutoPilot { (sender: ActorRef, msg: Any) =>
       msg match {
         case _: Message.GetSnapshot => sender ! Response.GetSnapshot(None)
+        case _ => throw new IllegalArgumentException(s"Unexpected error")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/actors/MatcherActorSpecification.scala:507]: Let's write the `msg` in the error message + RuntimeException

File: dex/src/test/scala/com/wavesplatform/dex/model/ExchangeTransactionCreatorSpecification.scala [hunk 9]
```diff
@@ -46,6 +46,7 @@ class ExchangeTransactionCreatorSpecification
 
             tc.createTransaction(oe).explicitGet() shouldBe a[ExchangeTransactionV2]
           }
+        case _ => throw new IllegalArgumentException(s"Unexpected list")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/model/ExchangeTransactionCreatorSpecification.scala:49]: `s` is not required + RuntimeException

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala [hunk 10]
```diff
@@ -104,6 +104,7 @@ class OrderValidatorSpecification
         val unsigned = newBuyOrder(pk) match {
           case x: OrderV1 => x.copy(amount = 0L)
           case x: OrderV2 => x.copy(amount = 0L)
+          case _ => throw new IllegalArgumentException(s"Invalid key pair")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala:107]: `throw new RuntimeException(s"Unexpected order: $x")`

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala [hunk 11]
```diff
@@ -117,6 +118,7 @@ class OrderValidatorSpecification
         val order = newBuyOrder(pk) match {
           case x: OrderV1 => x.copy(proofs = Proofs(Seq(ByteStr(Array.emptyByteArray))))
           case x: OrderV2 => x.copy(proofs = Proofs(Seq(ByteStr(Array.emptyByteArray))))
+          case _ => throw new IllegalArgumentException(s"Unexpected key pair")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala:121]: Same as above

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 2 above)
Comment (ngadiyak) [reply build.sbt:106]: without it fails on compile, i will try

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala (hunk 11 above)
Comment (ngadiyak) [reply dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala]: fixed

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala (hunk 10 above)
Comment (ngadiyak) [reply dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala]: fixed

File: dex/src/test/scala/com/wavesplatform/dex/model/ExchangeTransactionCreatorSpecification.scala (hunk 9 above)
Comment (ngadiyak) [reply dex/src/test/scala/com/wavesplatform/dex/model/ExchangeTransactionCreatorSpecification.scala]: fixed

File: dex/src/test/scala/com/wavesplatform/dex/actors/MatcherActorSpecification.scala (hunk 8 above)
Comment (ngadiyak) [reply dex/src/test/scala/com/wavesplatform/dex/actors/MatcherActorSpecification.scala]: fixed

File: dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala (hunk 7 above)
Comment (ngadiyak) [reply dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala]: fixed

File: dex-load/src/main/scala/com/wavesplatform/dex/load/request/Request.scala (hunk 6 above)
Comment (ngadiyak) [reply dex-load/src/main/scala/com/wavesplatform/dex/load/request/Request.scala]: fixed

File: dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala (hunk 5 above)
Comment (ngadiyak) [reply dex-load/src/main/scala/com/wavesplatform/dex/load/TankGenerator.scala]: fixed

File: dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsPingPongBaseSuite.scala (hunk 4 above)
Comment (ngadiyak) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsPingPongBaseSuite.scala:141]: ok, got it

File: dex-it/src/test/scala/com/wavesplatform/it/WsSuiteBase.scala (hunk 3 above)
Comment (ngadiyak) [reply dex-it/src/test/scala/com/wavesplatform/it/WsSuiteBase.scala:59]: fixed

File: build.sbt (hunk 2 above)
Comment (ngadiyak) [reply build.sbt:106]: `[error] sbt.librarymanagement.ResolveException: Error downloading org.scalameta:semanticdb-scalac_2.13.4:4.3.20`

File: dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala [hunk 12]
```diff
@@ -37,7 +37,7 @@ class WsCollectChangesClient(apiUri: String, address: String, aus: String, obs:
       }
       orderBookUpdates.put(diff.assetPair, updatedOb)
       none
-    case _ => throw new IllegalArgumentException(s"Unexpected message")
+    case x: Any => throw new IllegalArgumentException(s"Unexpected message: $x")
```
Comment (vsuharnikov) [inline dex-load/src/main/scala/com/wavesplatform/dex/load/ws/WsCollectChangesClient.scala:40]: `: Any` is not required

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala [hunk 13]
```diff
@@ -104,7 +104,7 @@ class OrderValidatorSpecification
         val unsigned = newBuyOrder(pk) match {
           case x: OrderV1 => x.copy(amount = 0L)
           case x: OrderV2 => x.copy(amount = 0L)
-          case _ => throw new IllegalArgumentException(s"Invalid key pair")
+          case x: Any => throw new RuntimeException(s"Unexpected order: $x")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala:107]: same

File: dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala [hunk 14]
```diff
@@ -118,7 +118,7 @@ class OrderValidatorSpecification
         val order = newBuyOrder(pk) match {
           case x: OrderV1 => x.copy(proofs = Proofs(Seq(ByteStr(Array.emptyByteArray))))
           case x: OrderV2 => x.copy(proofs = Proofs(Seq(ByteStr(Array.emptyByteArray))))
-          case _ => throw new IllegalArgumentException(s"Unexpected key pair")
+          case x: Any => throw new RuntimeException(s"Unexpected order: $x")
```
Comment (vsuharnikov) [inline dex/src/test/scala/com/wavesplatform/dex/model/OrderValidatorSpecification.scala:121]: same

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 2 above)
Comment (vsuharnikov) [reply build.sbt]: It helped?

Comment (vsuharnikov) [reply build.sbt]: Ah see. You need to update:
```
"ch.epfl.scala" % "sbt-scalafix" % "0.9.19",
```
in `plugins.sbt`.

Comment (ngadiyak) [reply build.sbt]: ok, tnx

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**